lib/Doctrine/ORM/EntityManager.php

```diff
@@ -157,8 +156,12 @@ protected function __construct(Connection $conn, Configuration $config, EventMan
         $this->metadataFactory->setCacheDriver($this->config->getMetadataCacheImpl());
 
         $this->repositoryFactory = $config->getRepositoryFactory();
-        $this->unitOfWork        = new UnitOfWork($this);
-        $this->proxyFactory      = new ProxyFactory(
+
+        $unitOfWorkClassName  = $config->getUnitOfWorkClassName();
+        $this->unitOfWork     = new $unitOfWorkClassName($this);
```

Comment:
please don't do this. Configuring a class name is a bad extension point, because it assumes what the dependencies of the class are. If we make it configurable, we should do it cleanly. thus, this is a totally different topic than making ``EntityManager::__construct`` public, so it should be submitted separately (I'm against making the UnitOfWork class configurable btw, so I would vote for rejecting this PR entirely because of this change given that you are mixing things)